### ISSUE TITLE
[Circles] admin should be able to edit some more settings 

### ISSUE THREAD

**Issue by ArtificialOwl** [nextcloud/contacts#2644]
**Is your feature request related to a problem? Please describe.**

With the last version of Circles, we are adding some customization to circles.
This allow admins of a Circle to:
  - enforce password protection on link generated when sharing files to external members.
  - allow the creation of an optional single password to protect those shares. (If not, a random password will be generated)

**Describe the solution you'd like**

Those settings needs to be configurable by an admin and can be displayed in the same page than the current page when an admin can set the visibility of your circle.
If on the same page, it needs to be fully separated from the current 'config' part of the Circle.

**Describe alternatives you've considered**

Requesting help from @nextcloud/designers :)

**Additional context**

Settings should already be available as an array in the model Circle, and returned by the back-end when a request includes a Circle.
Example of curl request to update a setting:
```
$ curl -X PUT "http://nc23.local/ocs/v2.php/apps/circles/circles/GrjA5QR2J8trwyYEBUjy1tOdyGVZUuj/setting?format=json" -H "OCS-APIRequest: true" -u "user:password" -d setting=test -d value=123
{
  "ocs": {
    "meta": {
      "status": "ok",
      "statuscode": 200,
      "message": "OK"
    },
    "data": {
      "id": "GrjA5QR2J8trwyYEBUjy1tOdyGVZUuj",
      "name": "t001",
      "displayName": "t001",
      "sanitizedName": "t001",
      "source": 16,
      "population": 0,
      "config": 0,
      "description": "",
      "settings": {
        "ouila": "123",
        "test": "123"
      },
  [...]
```

If value is not provided, setting will be removed.

**Comment by ArtificialOwl** [comment]
current PR on back-end: https://github.com/nextcloud/circles/pull/926

**Comment by PVince81** [comment]
@ChristophWurst

**Comment by ChristophWurst** [comment]
yes?

**Comment by ChristophWurst** [comment]
> **Describe alternatives you've considered**
> 
> Requesting help from @nextcloud/designers :)

Friendly ping @jancborchardt @nimishavijay. Please let us know how this should work from a user's perspective.

**Comment by jancborchardt** [comment]
@ArtificialOwl I'm not entirely sure I understand. Do you have a screenshot of how it looked like in the previous version of Circles (the one not in Contacts yet)?

**Comment by ArtificialOwl** [comment]
Hello,

Let me show you 2 screenshots:  


### The old circle's setting page `<NC22`
![Selection_034](https://user-images.githubusercontent.com/1038617/170017718-badc6555-d8d6-4247-9bc9-33e281d4a8c2.png)

### The new one `>= NC22`
![Selection_035](https://user-images.githubusercontent.com/1038617/170017728-05e141e1-8bc6-4981-b666-62ff4e7b9f12.png)

The new settings page of a Circle is missing some entries:

- `enable password protection` on all files shared to the Circle
- `enable single password protection` so that only one password will be generated for all files shared to the Circle.
- `set single password` to define the single password to be used when sharing a file to the Circle.

Note that **enable single password protection** and **set single password** are linked: if enable is true but no single password is set, a random password will be generated for each new shares.

In my opinion those new entries in the setting page needs to be added just before the 'Delete Circle' button, slightly separated from the current available options.

**Comment by ArtificialOwl** [comment]
We can arrange a call to discuss about some missing details !

**Comment by artonge** [comment]
Reopening as it seem to still occure.

**Comment by mejo-** [comment]
@artonge not sure what you meant with "seems to still occur". The issue at hand seems to be fixed with #2766, no?

**Comment by artonge** [comment]
@mejo-, it's been more than a year since that comment, I don't remember the background, sorry. Feel free to ignore it.
